src/execution/error.rs

```diff
@@ -14,7 +14,7 @@ pub enum RuntimeError {
     InvalidImportType,
     UnknownImport,
     MoreThanOneMemory,
-    OutOfFuel,
+    OutOfFuel { required_fuel: u32 },
```

Comment:
We should not make this a public error, the user will never see it:

either there is no fuel enabled, or an out-of-fuel event will trigger a resumable to be returned to the user.